### DISCUSSION TITLE
ERROR of building: Unknown module(s) in QT: 3dinput

### DISCUSSION THREAD

**Discussion by NailGit** [ahrm/sioyek#816] [General]
OS: Linux Mint Mate 21.  I made a compilation with the error: Project ERROR: Unknown module(s) in QT: 3dinput . 
I installed libqt53dinput5 but with the same error. How do you think I could solve this problem?

**Comment by ahrm** [comment]
I can't be sure these issues are very distribution-specific. Are you sure you are not using a different version of qt for compilation?

**Comment by NailGit** [comment]
The version QT is 5.15.1:
`qmake --version

QMake version 3.1
Using Qt version 5.15.3 in /usr/lib/x86_64-linux-gnu`

**Reply by ahrm** [reply]
It seems like you are using two different qt versions, make sure to install 3dinput for the qt version you are using to compile sioyek.

**Reply by NailGit** [reply]
Maybe. But I installed libqt53dinput5 which is 3dinput for Qt5.

**Reply by NailGit** [reply]
I found a solution. I installed qt3d5-dev and the compilation is finished without error.

**Reply by ice7512** [reply]
This worked for me too. I am Linux Mint 22.2 Cinnamon. I also had to execute the following commands to persuade the script to build successfully (in an active virtual environment):
pip install PySide6
pip install PyQt6
pip install PyQt5
sudo apt-get install qt5-qmake
sudo apt-get install qtbase5-dev
sudo apt-get install libglfw3-dev libgl1-mesa-dev libglu1-mesa-dev

**Reply by NailGit** [reply]
Hi, 
Thanks for the message! I have discussed this question once more. After installing QT6, some programs that I use every day stopped working. For this reason, I uninstalled QT6 and stopped any further compilation. I have used Evince for many years, and it is sufficient for my needs. I'll try installing only what appears in the repository.

**Reply by ice7512** [reply]
My big question is actually if anyone is using a Linux Distro that has Qt 6.7.3 or greater installed system-wide anywhere at all. It seems impossible to install it without causing things to break. @ahrm did you only successfully compile on Fedora?

**Reply by ice7512** [reply]
I just figured out how to compile it [here](https://github.com/ahrm/sioyek/issues/1517) if you would still like to try it out. I got smooth auto scroll on it and now I have everything I want in one place essentially.

**Reply by NailGit** [reply]
Thank you, but I will wait for the official version in the repository. Evince meets all my needs.
